frontend/src/components/metadata/CreateMetadataDefinition.tsx

```diff
@@ -0,0 +1,307 @@
+										<Grid xs={4} md={4}>
+											<FormGroup>
+												<FormControlLabel control={<Checkbox 
+													checked={input.list}
+													onChange={(event) => { handleInputChange(index, "list", event.target.value); }}
+													/>} label="Contains List" />
+												<FormControlLabel control={<Checkbox 
+													checked={input.required}
+													onChange={(event) => { handleInputChange(index, "required", event.target.value); }}
+													/>} label="Required" />
+											</FormGroup>
+										</Grid>
+									</Grid>
+									<TextField
+										variant="outlined"
+										margin="normal"
+										required
+										fullWidth
+										id="field-name"
+										label="Field Name"
+										placeholder="Please enter field name"
+										value={input.name}
+										onChange={(event) => { handleInputChange(index, "name", event.target.value); }}
+									/>
+									<FormControl fullWidth>
+										<InputLabel id="widget-type-select">Select Widget Type</InputLabel>
+										<Select
+											labelId="widget-type-select-label"
+											id="widget-type-select-label"
+											value={input.widgetType}
+											onChange={(event) => { handleInputChange(index, "widgetType", event.target.value); }}
+											label="widgetType">
+											{widgetTypes.map((input, index) => {
+												return (
+													<MenuItem value={input}>{input}</MenuItem>
+												);
+											})}
+										</Select>
+									</FormControl>
+									<TextField
+										variant="outlined"
+										margin="normal"
+										required
+										fullWidth
+										id="datatype"
+										label="Field Data Type"
+										placeholder="Please enter field data type (e.g. int, str, enum)"
+										name="Field Data Type"
+										value={input.config.type}
+										onChange={(event) => { handleInputChange(index, "type", event.target.value); }}
+									/>
+									{(input.config.type == "enum") ? <>
+										<TextField
+											variant="outlined"
+											margin="normal"
+											required
+											fullWidth
+											id="options"
+											label="options"
+											name="Field Data Options"
+											multiline
+											maxRows={6}
+										/>
+									</> : <></>}
+									{index != formInput.fields.length - 1 ? <Button  variant="contained" onClick={handleNext}>Next</Button> : <Button  variant="contained" onClick={handleNext}>Finish</Button> }
+									<Button onClick={handleBack}>Back</Button>
+								</StepContent>
+							</Step>
+						);
+					})}
+					
+					<Step key="submit">
+						<StepLabel>Submit</StepLabel>
+						<StepContent>
+							<Button
+								variant="contained"
+								onClick={postMetadata}
+								sx={{ mt: 1, mr: 1, "alignItems": "right" }}>
+								Submit
+							</Button>
+							<Button
+								onClick={clearForm}
+								sx={{ mt: 1, mr: 1, "alignItems": "right" }}>
+								Clear
+							</Button>
```

Comment:
IMO it would be nice to add a "Back" at the very last step too. Sometimes I still want to change things until the very last...

Reply:
I've updated the stepper code, now you can click directly on the stepper number to navigate back. I figured that a dding too many buttons would make the UI cluttered, hence adopted this approach instead :)